code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -960,13 +976,15 @@
 /mob/living/silicon/robot/proc/radio_menu()
 	radio.interact(src) //Just use the radio's Topic() instead of bullshit special-snowflake code
 
+
 /mob/living/silicon/robot/Move(a, b, flag)
 	. = ..()
 
 	if(module)
 		if(module.type == /obj/item/robot_module/janitor)
+			var/obj/item/robot_module/janitor/J = module
 			var/turf/tile = get_turf(src)
-			if(isturf(tile))
+			if(isturf(tile)&& J.mopping == 1)
```

Comment:
```suggestion
			if(isturf(tile) && J.mopping)
```
No need to do the == for TRUE/FALSE variables.
If not mopping, J.mopping will be 0 (FALSE), and will return as such. 
So J.mopping is the same as J.mopping == 1, and !J.mopping is the same as J.mopping == 0